core/src/main/java/org/apache/calcite/sql/SqlJsonTypeNameSpec.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.util.Litmus;
+
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/**
+ * A sql type name specification of the JSON type.
+ */
+public class SqlJsonTypeNameSpec extends SqlTypeNameSpec {
+
+  public final Integer maxLength;
+  public final Integer inlineLength;
+  public final CharacterSet characterSet;
+  public final StorageFormat storageFormat;
+
+  public SqlJsonTypeNameSpec(SqlParserPos pos,
+      Integer maxLength,
+      Integer inlineLength,
+      CharacterSet characterSet,
+      StorageFormat storageFormat) {
```

Comment:
nit: indentation looks a little off?

Reply:
It seems to looks consistent with other classes that I looked at such as [SqlCreateTable](https://github.com/googleinterns/calcite/blob/16820a37f33870fdc36d37e8683c511d01e30207/core/src/main/java/org/apache/calcite/sql/SqlCreateTable.java#L83). How do you think it should look like?